src/components/ProductTour/ProductTour.tsx

```diff
@@ -0,0 +1,155 @@
+        setIsChecked(!isChecked);
+    };
+
+    const handleDotClick = (i: number) => {
+        setCurrentSlide(i);
+    };
+
+    const handleDotKeyDown = (e: React.KeyboardEvent<HTMLElement>, i: number) => {
+        if (e.key === ENTER_KEY) {
+            handleDotClick(i);
+        }
+    };
+
+    const handleCancel = () => {
+        onCancel?.(isChecked);
+    };
+
+    const handleNext = () => {
+        setCurrentSlide(currentSlide + 1);
+    };
+
+    const handleFinish = () => {
+        onFinished(isChecked);
+    };
+
+    const handledOverlayClick = () => {
+        if (isLastSlide) {
+            handleFinish();
+        } else {
+            handleCancel();
+        }
+    };
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            isPositionFixed
+            {...rest}
+            {...block()}
+            onRequestClose={handledOverlayClick}
+        >
+            <section {...elem('slide')}>
+                {nOfSlides === 1 ? children : children[currentSlide]}
+            </section>
+            <div {...elem('footer')}>
+                <div {...elem('checkbox')}>
+                    {checkboxLabel && (
+                        <Checkbox id="hide" checked={isChecked} onChange={handleCheckboxClick}>
+                            {checkboxLabel}
+                        </Checkbox>
+                    )}
+                </div>
+                {nOfSlides > 1 && (
+                    <div>
+                        {[...Array(nOfSlides)].map((v, i) => {
+                            const isNotActive = i !== currentSlide;
+                            return (
+                                <div
+                                    key={i} // eslint-disable-line react/no-array-index-key
+                                    role="button"
+                                    tabIndex={isNotActive ? 0 : undefined}
+                                    onClick={isNotActive ? () => handleDotClick(i) : undefined}
+                                    onKeyDown={
+                                        isNotActive ? (e) => handleDotKeyDown(e, i) : undefined
+                                    }
+                                    {...elem('dot', { isActive: !isNotActive })}
+                                />
+                            );
+                        })}
+                    </div>
+                )}
+                <div {...elem('buttons')}>
+                    {isLastSlide ? (
+                        <Button context="primary" onClick={handleFinish}>
+                            {finishLabel}
+                        </Button>
+                    ) : (
+                        <>
+                            <Button context="link" onClick={handleCancel}>
+                                {cancelLabel || ''}
+                            </Button>
+                            <Button context="brand" onClick={handleNext}>
+                                {continueLabel || ''}
```

Comment:
I think these two label values can be set as default parameters during `props` object deconstruction